plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/Ros2CodeGenerator.xtend

```diff
@@ -248,4 +280,68 @@ int main(int argc, char * argv[])
  def String check_name(String interface_name){
  	return interface_name.replace("/","");
  }
-}
\ No newline at end of file
+ 
+ def Boolean is_array(EList<ParameterType> list) {
+ 	var t = list.get(0).class
+ 	for (type : list) {
+ 		if (!(type.class == t)) {
+ 			return false;
+ 		}
+ 	}
+    return true;
+ }
+ 
+ //TODO: should get resolved name from ParameterStructTypeMember
+ def String compile_struct(ParameterStructTypeImpl struct) {
+	var struct_str = "";
+  	for (elem : struct.getParameterstructypetmember()) {
+  		if (elem.getType() instanceof ParameterStructTypeImpl) {
+  			struct_str += compile_struct(elem.getType() as ParameterStructTypeImpl);
+  		} else {
+	  		var param_pair = elem.getType().get_param_type();
+  			struct_str += "this->declare_parameter<" + param_pair.getKey() + ">(\"" + elem.getName() + param_pair.getValue() + "\");\n";
+//  			struct_str += "this->get_parameter(\"" + elem.getName() + "\", " + elem.getName() + "_);\n\n";
+  		}
+  	}
+	return struct_str;
+ }
+
+ def Pair<String, String> get_param_type(ParameterType type) {
```

Comment:
is it possible to move "get_param_type" to another class, called for example "code_generator_helper" that can be used by the ROS2 and the ROS1 code generator?

Similar to https://github.com/ipa320/ros-model/blob/master/plugins/de.fraunhofer.ipa.rossystem.xtext/src/de/fraunhofer/ipa/rossystem/generator/GeneratorHelpers.xtend